--- HelloWorld.s	2018-02-13 16:05:24.927132500 +0800
+++ HelloWorld-gcc.s	2018-02-13 16:08:39.850281400 +0800
@@ -1,126 +1,115 @@
-	.text
-	.def	 @feat.00;
-	.scl	3;
-	.type	0;
-	.endef
-	.globl	@feat.00
-@feat.00 = 1
-	.def	 ___cxx_global_var_init;
-	.scl	3;
-	.type	32;
-	.endef
-	.p2align	4, 0x90         # -- Begin function __cxx_global_var_init
-___cxx_global_var_init:                 # @__cxx_global_var_init
-	.cfi_startproc
-# BB#0:
-	pushl	%ebp
-Lcfi0:
-	.cfi_def_cfa_offset 8
-Lcfi1:
-	.cfi_offset %ebp, -8
-	movl	%esp, %ebp
-Lcfi2:
-	.cfi_def_cfa_register %ebp
-	subl	$8, %esp
-	leal	__ZStL8__ioinit, %ecx
-	calll	__ZNSt8ios_base4InitC1Ev
-	leal	___dtor__ZStL8__ioinit, %ecx
-	movl	%ecx, (%esp)
-	calll	_atexit
-	movl	%eax, -4(%ebp)          # 4-byte Spill
-	addl	$8, %esp
-	popl	%ebp
-	retl
-	.cfi_endproc
-                                        # -- End function
-	.def	 ___dtor__ZStL8__ioinit;
-	.scl	3;
-	.type	32;
-	.endef
-	.p2align	4, 0x90         # -- Begin function __dtor__ZStL8__ioinit
-___dtor__ZStL8__ioinit:                 # @__dtor__ZStL8__ioinit
-	.cfi_startproc
-# BB#0:
-	pushl	%ebp
-Lcfi3:
-	.cfi_def_cfa_offset 8
-Lcfi4:
-	.cfi_offset %ebp, -8
-	movl	%esp, %ebp
-Lcfi5:
-	.cfi_def_cfa_register %ebp
-	leal	__ZStL8__ioinit, %ecx
-	calll	__ZNSt8ios_base4InitD1Ev
-	popl	%ebp
-	retl
-	.cfi_endproc
-                                        # -- End function
-	.def	 _main;
-	.scl	2;
-	.type	32;
-	.endef
-	.globl	_main                   # -- Begin function main
-	.p2align	4, 0x90
-_main:                                  # @main
-	.cfi_startproc
-# BB#0:
-	pushl	%ebp
-Lcfi6:
-	.cfi_def_cfa_offset 8
-Lcfi7:
-	.cfi_offset %ebp, -8
-	movl	%esp, %ebp
-Lcfi8:
-	.cfi_def_cfa_register %ebp
-	subl	$16, %esp
-	calll	___main
-	leal	__ZSt4cout, %eax
-	leal	L_.str, %ecx
-	movl	%eax, (%esp)
-	movl	%ecx, 4(%esp)
-	calll	__ZStlsISt11char_traitsIcEERSt13basic_ostreamIcT_ES5_PKc
-	leal	__ZSt4endlIcSt11char_traitsIcEERSt13basic_ostreamIT_T0_ES6_, %ecx
-	movl	%ecx, -4(%ebp)          # 4-byte Spill
-	movl	%eax, %ecx
-	movl	-4(%ebp), %eax          # 4-byte Reload
-	movl	%eax, (%esp)
-	calll	__ZNSolsEPFRSoS_E
-	subl	$4, %esp
-	xorl	%ecx, %ecx
-	movl	%eax, -8(%ebp)          # 4-byte Spill
-	movl	%ecx, %eax
-	addl	$16, %esp
-	popl	%ebp
-	retl
-	.cfi_endproc
-                                        # -- End function
-	.def	 __GLOBAL__sub_I_HelloWorld.cpp;
-	.scl	3;
-	.type	32;
-	.endef
-	.p2align	4, 0x90         # -- Begin function _GLOBAL__sub_I_HelloWorld.cpp
-__GLOBAL__sub_I_HelloWorld.cpp:         # @_GLOBAL__sub_I_HelloWorld.cpp
-	.cfi_startproc
-# BB#0:
-	pushl	%ebp
-Lcfi9:
-	.cfi_def_cfa_offset 8
-Lcfi10:
-	.cfi_offset %ebp, -8
-	movl	%esp, %ebp
-Lcfi11:
-	.cfi_def_cfa_register %ebp
-	calll	___cxx_global_var_init
-	popl	%ebp
-	retl
-	.cfi_endproc
-                                        # -- End function
-	.lcomm	__ZStL8__ioinit,1       # @_ZStL8__ioinit
-	.section	.rdata,"dr"
-L_.str:                                 # @.str
-	.asciz	"!!!Hello World!!!"
-
-	.section	.ctors,"dw"
-	.p2align	2
-	.long	__GLOBAL__sub_I_HelloWorld.cpp
-
+	.file	"HelloWorld.cpp"
+	.section .rdata,"dr"
+__ZStL19piecewise_construct:
+	.space 1
+.lcomm __ZStL8__ioinit,1,1
+	.def	___main;	.scl	2;	.type	32;	.endef
+LC0:
+	.ascii "!!!Hello World!!!\0"
+	.text
+	.globl	_main
+	.def	_main;	.scl	2;	.type	32;	.endef
+_main:
+LFB1546:
+	.cfi_startproc
+	leal	4(%esp), %ecx
+	.cfi_def_cfa 1, 0
+	andl	$-16, %esp
+	pushl	-4(%ecx)
+	pushl	%ebp
+	.cfi_escape 0x10,0x5,0x2,0x75,0
+	movl	%esp, %ebp
+	pushl	%ecx
+	.cfi_escape 0xf,0x3,0x75,0x7c,0x6
+	subl	$20, %esp
+	call	___main
+	movl	$LC0, 4(%esp)
+	movl	$__ZSt4cout, (%esp)
+	call	__ZStlsISt11char_traitsIcEERSt13basic_ostreamIcT_ES5_PKc
+	movl	$__ZSt4endlIcSt11char_traitsIcEERSt13basic_ostreamIT_T0_ES6_, (%esp)
+	movl	%eax, %ecx
+	call	__ZNSolsEPFRSoS_E
+	subl	$4, %esp
+	movl	$0, %eax
+	movl	-4(%ebp), %ecx
+	.cfi_def_cfa 1, 0
+	leave
+	.cfi_restore 5
+	leal	-4(%ecx), %esp
+	.cfi_def_cfa 4, 4
+	ret
+	.cfi_endproc
+LFE1546:
+	.def	___tcf_0;	.scl	3;	.type	32;	.endef
+___tcf_0:
+LFB2036:
+	.cfi_startproc
+	pushl	%ebp
+	.cfi_def_cfa_offset 8
+	.cfi_offset 5, -8
+	movl	%esp, %ebp
+	.cfi_def_cfa_register 5
+	subl	$8, %esp
+	movl	$__ZStL8__ioinit, %ecx
+	call	__ZNSt8ios_base4InitD1Ev
+	leave
+	.cfi_restore 5
+	.cfi_def_cfa 4, 4
+	ret
+	.cfi_endproc
+LFE2036:
+	.def	__Z41__static_initialization_and_destruction_0ii;	.scl	3;	.type	32;	.endef
+__Z41__static_initialization_and_destruction_0ii:
+LFB2035:
+	.cfi_startproc
+	pushl	%ebp
+	.cfi_def_cfa_offset 8
+	.cfi_offset 5, -8
+	movl	%esp, %ebp
+	.cfi_def_cfa_register 5
+	subl	$24, %esp
+	cmpl	$1, 8(%ebp)
+	jne	L6
+	cmpl	$65535, 12(%ebp)
+	jne	L6
+	movl	$__ZStL8__ioinit, %ecx
+	call	__ZNSt8ios_base4InitC1Ev
+	movl	$___tcf_0, (%esp)
+	call	_atexit
+L6:
+	nop
+	leave
+	.cfi_restore 5
+	.cfi_def_cfa 4, 4
+	ret
+	.cfi_endproc
+LFE2035:
+	.def	__GLOBAL__sub_I_main;	.scl	3;	.type	32;	.endef
+__GLOBAL__sub_I_main:
+LFB2037:
+	.cfi_startproc
+	pushl	%ebp
+	.cfi_def_cfa_offset 8
+	.cfi_offset 5, -8
+	movl	%esp, %ebp
+	.cfi_def_cfa_register 5
+	subl	$24, %esp
+	movl	$65535, 4(%esp)
+	movl	$1, (%esp)
+	call	__Z41__static_initialization_and_destruction_0ii
+	leave
+	.cfi_restore 5
+	.cfi_def_cfa 4, 4
+	ret
+	.cfi_endproc
+LFE2037:
+	.section	.ctors,"w"
+	.align 4
+	.long	__GLOBAL__sub_I_main
+	.ident	"GCC: (Rev2, Built by MSYS2 project) 7.2.0"
+	.def	__ZStlsISt11char_traitsIcEERSt13basic_ostreamIcT_ES5_PKc;	.scl	2;	.type	32;	.endef
+	.def	__ZSt4endlIcSt11char_traitsIcEERSt13basic_ostreamIT_T0_ES6_;	.scl	2;	.type	32;	.endef
+	.def	__ZNSolsEPFRSoS_E;	.scl	2;	.type	32;	.endef
+	.def	__ZNSt8ios_base4InitD1Ev;	.scl	2;	.type	32;	.endef
+	.def	__ZNSt8ios_base4InitC1Ev;	.scl	2;	.type	32;	.endef
+	.def	_atexit;	.scl	2;	.type	32;	.endef
